tests/openssh.rs

```diff
@@ -282,324 +350,345 @@ macro_rules! assert_kind {
+
+        let status = dbg!(session
+            .raw_command("printf '%d %d'")
+            .arg("1")
+            .arg("2")
+            .output()
+            .await
+            .unwrap())
+        .status;
+        assert!(status.success());
 
-    session.close().await.unwrap();
+        session.close().await.unwrap();
+    }
 }
 
 #[tokio::test]
 #[cfg_attr(not(ci), ignore)]
 async fn process_exit_on_signal() {
-    let session = Session::connect(&addr(), KnownHosts::Accept).await.unwrap();
-
-    let mut sleeping = session.command("sleep").arg("5566").spawn().unwrap();
-
-    // give it some time to make sure it starts
-    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-
-    // Now stop that process.
-    //
-    // We use `pkill -f` to match on the number rather than the `sleep` command, since other tests
-    // may use `sleep`. We use `-o` to ensure that we don't accidentally kill the ssh connection
-    // itself, but instead match the _oldest_ matching command.
-    let killed = session
-        .command("pkill")
-        .arg("-f")
-        .arg("-o")
-        .arg("5566")
-        .output()
-        .await
-        .unwrap();
-    eprintln!("{:?}", killed);
-    assert!(killed.status.success());
-
-    // await that process — this will yield "Disconnected", since the remote process disappeared
-    let failed = sleeping.wait().await.unwrap_err();
-    eprintln!("{:?}", failed);
-    assert!(matches!(failed, Error::Disconnected));
-
-    // the connection should still work though
-    let _ = session.check().await.unwrap();
+    for session in connects().await {
+        let mut sleeping = session.command("sleep").arg("5566").spawn().await.unwrap();
+
+        // give it some time to make sure it starts
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+        // Now stop that process.
+        //
+        // We use `pkill -f` to match on the number rather than the `sleep` command, since other tests
+        // may use `sleep`. We use `-o` to ensure that we don't accidentally kill the ssh connection
+        // itself, but instead match the _oldest_ matching command.
+        let killed = session
+            .command("pkill")
+            .arg("-f")
+            .arg("-o")
+            .arg("5566")
+            .output()
+            .await
+            .unwrap();
+        eprintln!("process_exit_on_signal: {:?}", killed);
+        assert!(killed.status.success());
+
+        // await that process — this will yield "Disconnected", since the remote process disappeared
+        eprintln!("process_exit_on_signal: Waiting for sleeping to exit");
+        let failed = sleeping.wait().await.unwrap_err();
+        eprintln!("process_exit_on_signal: {:?}", failed);
+        assert!(matches!(failed, Error::RemoteProcessTerminated));
+
+        // the connection should still work though
+        let _ = session.check().await.unwrap();
+    }
 }
 
+#[cfg(not(feature = "mux_client"))]
```

Comment:
This doesn't seem right. This will not run the given test for _any_ backend when the feature flag is set.

Reply:
Fixed.